### PR TITLE
[skip ci] Add segment classification, statistical relevance tests

### PR DESCRIPTION
**Pull request by JacopoTeneggi**
This PR adds:

- Segment classification based on the depth of the node in the spline tree. We classify a segment as `axon` if it is the root of the spline tree, `terminal` if it does not have any successors, `collateral` otherwise.
- Updated distribution, scatter, and regression figures with the new classification (with `seaborn`).
- Kolgomorov-Smirnoff and Wilcoxon sum rank (Mann-Whitney U) tests to quantify the statistical difference between the curvature/torsion distributions between the three classes.

### CODE REVIEW

File: experiments/axon_geometry/scripts/generate_trace_data.py [hunk 1]
```diff
@@ -45,11 +56,65 @@ def generate_brain_trace_data(brain: str, spacing: int):
             print("Initialized GeometricGraph")
             spline_tree = G.fit_spline_tree_invariant()
             print("Computed splines")
+
```
Comment (tathey1) [inline experiments/axon_geometry/scripts/generate_trace_data.py:59]: Why is all this code commented? If you want this code to stay, is it possible to put it inside a conditional? If its not necessary, can you remove?

**Review by tathey1 — CHANGES_REQUESTED** [review]
Can you add a comment on this PR what the changes you made to [this](https://github.com/neurodata/brainlit/blob/jaco-newfigs/experiments/axon_geometry/notebooks/linear_regression_from_root.ipynb) and [this](https://github.com/neurodata/brainlit/blob/jaco-newfigs/experiments/axon_geometry/notebooks/linear_regression_length_segment.ipynb) file?

**Comment by JacopoTeneggi** [comment]
Changes to https://github.com/neurodata/brainlit/blob/jaco-newfigs/experiments/axon_geometry/notebooks/linear_regression_from_root.ipynb :

- Clear outputs

Changes to https://github.com/neurodata/brainlit/blob/jaco-newfigs/experiments/axon_geometry/notebooks/linear_regression_length_segment.ipynb :

- Clear outputs
- Remove pandas.DataFrame structure to save data
- Remove seaborn plot cell

File: experiments/axon_geometry/scripts/generate_trace_data.py (hunk 1 above)
Comment (JacopoTeneggi) [reply experiments/axon_geometry/scripts/generate_trace_data.py]: Not necessary, removed. It was the code I used to plot the dendograms of the spline trees

**Review by tathey1 — APPROVED** [review]
Looks good, go ahead and run the checks again and I'll merge it.

**Pull request merged**